scripts/generate-contract-bindings.sh

```
@@ -75,6 +80,20 @@ echo -e "${GREEN}========================================${NC}"
 echo -e "${GREEN}Bindings generated successfully!${NC}"
 echo -e "${GREEN}========================================${NC}"
 echo ""
+
+if [ "$SKIP_TS_CHECK" != "true" ] && [ -d "frontend" ] && [ -f "frontend/tsconfig.json" ]; then
+    echo -e "${YELLOW}Validating frontend TypeScript compilation...${NC}"
+    if command -v npx &> /dev/null; then
+        (cd frontend && npx tsc --noEmit)
+        if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)